### ISSUE TITLE
Freeze installed chart version in HelmRelease

### ISSUE THREAD

**Issue by raffis** [fluxcd/helm-controller#212]
I think this is not yet possible so here is my use case:

I'd like to install a couple of HelmReleases with a semver expr of `*` or any other version selector.
Meaning I'd like to install the latest published chart.
However I need a flag to freeze the version once selected and installed because I want avoid automated updates of these releases if it gets reconciled.
In other words I want to install the latest version but avoid further automated updates.

Basically a flag would be required on the HelmRelease, something like `freezeVersion bool`.

Happy to create a pr for this if you can imagine merging this.

**Comment by hiddeco** [comment]
What prevents you from just pinning it to the right version?

Asking because I can see this all go wrong when doing e.g. a disaster recovery, where the dependency has moved on but your values (that are basically married to whatever chart you make use of) still think an older version applies.

**Comment by raffis** [comment]
> What prevents you from just pinning it to the right version?

Because the version is not necessarily known at that point. Lets say we have a big kustomize base with x helmreleases of internally developed services, if somebody installs that base into a new environment the should get the latest stable releases. 

> Asking because I can see this all go wrong when doing e.g. a disaster recovery, where the dependency has moved on but your values (that are basically married to a version of whatever chart you make use of) still think an older version applies.

That's actually a good point, another way might be be to let the reconciler write the elected chart version to its own spec if freezeVersion is set. (What the default is doing with `*` if no version is set).

**Comment by raffis** [comment]
> That's actually a good point, another way might be be to let the reconciler write the elected chart version to its own spec if freezeVersion is set. (What the default is doing with `*` if no version is set).

Mhm this would only work if no version was set.  Actually I would just need a way to change the default behavior of setting `*` as spec.chart.version. In my case it should set the elected version.

**Comment by hiddeco** [comment]
> That's actually a good point, another way might be be to let the reconciler write the elected chart version to its own spec if freezeVersion is set. (What the default is doing with `*` if no version is set).

If it would write it to its own (in-cluster) spec this would not make any difference in case of a disaster recovery? (Assuming Flux2 is used, and the source of truth is in Git).

> Actually I would just need a way to change the default behavior of setting * as spec.chart.version. In my case it should set the elected version.

Can't this be done by using an "elected version" Kustomize overlay that is kept in sync together with the values?

**Comment by raffis** [comment]
> Can't this be done by using an "elected version" Kustomize overlay that is kept in sync together with the values?

Reckon something like this will be the alternative yes.

**Comment by hiddeco** [comment]
Due to the reason provided in https://github.com/fluxcd/helm-controller/issues/212#issuecomment-775813136, we are not planning on supporting this.
